.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,48 @@
+name: Publish to NPM
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploy-sha:
+        description: 'SHA of the commit to deploy'
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      deploy-sha:
+        description: 'SHA of the commit to deploy'
+        required: true
+        type: string
+  
+jobs:
+  publish-to-npm:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.deploy-sha }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.12.x
+          registry-url: https://registry.npmjs.org/
+          scope: '@constructorio'
+          token: ${{ secrets.NPM_TOKEN }}
+      - name: Install dependencies
+        run: npm ci
+      - name: Build compiled package for publishing
+        run: npm run compile
+      - name: Publish to NPM
+        run: npm publish --access public
```

Comment:
Currently using `--dry-run` on the fork repository main branch to avoid real publishes from triggering here, so don't worry about testing there making real publishes. Only this branch has it in live form.

Flagging this as a specific area of interest for the CI team to validate

Reply:
very thoughtful setup for testing, thanks for calling this out @KagenLH  :)